dev_guide/builds.adoc

```diff
@@ -191,6 +191,84 @@ See the xref:../creating_images/s2i.adoc#creating-images-s2i[S2I Requirements] t
 on how to create a builder image supporting incremental builds.
 ====
 
+
+[[extended-builds]]
+
+=== Extended builds
+
+For compiled languages (Go, C, C++, Java, etc.) the dependencies necessary for
+compilation might increase the size of the image or introduce vulnerabilities
+that can be exploited.
+
+To avoid these problems, source-to-image introduces a two image build process
+that allows an application to be built via the normal flow in a builder image
+but then injects the resulting application artifacts into a runtime-only image
+for execution.
+
+To offer flexibility in this process, source-to-image will execute an
+*_assemble-runtime_* script inside the runtime image that allows further
```

Comment:
@ahardin-rh aha! thank you for looking into it, will address all the below comments and correct it.